### PR TITLE
Feature (#31995): Echo Core - Base Error 

### PR DESCRIPTION
**Pull request by theVoogie**
Moved BaseError and NetworkError to the EchoCore.

### CODE REVIEW

File: src/errors/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * ErrorProperties provides a general purpose object
+ * for extending the number error porperties available
+ * @export
+ * @interface ErrorProperties
+ */
+export interface ErrorProperties {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    errors?: any;
```
Comment (Noggling) [inline src/errors/types.ts:9]: kan vi heller bruke unknown her?

Comment (Noggling) [reply src/errors/types.ts:9]: mulig vi kan bruke string | Record<string, unknown>

File: src/errors/types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ * ErrorProperties provides a general purpose object
+ * for extending the number error porperties available
+ * @export
+ * @interface ErrorProperties
+ */
+export interface ErrorProperties {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    errors?: any;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any; //support any properties in exception/error
```
Comment (Noggling) [inline src/errors/types.ts:11]: samme som over ;)

File: src/errors/BaseError.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @param exception a generated exception provided as a {Record<string, unknown>} type
+ * @export
+ * @class BaseError
+ * @extends {Error}
+ */
+export default class BaseError extends Error {
+    protected properties: ErrorProperties;
+    hasBeenLogged: boolean;
+
+    constructor(message: string, exception: Record<string, unknown>) {
```
Comment (OveBDev) [inline src/errors/BaseError.ts:16]: Exception should be optional, you can take a look in echopedia where this already has been done

**Review by hkot1991 — APPROVED** [review]

File: src/errors/types.ts (hunk 1 above)
Comment (theVoogie) [reply src/errors/types.ts:9]: ❤️

**Review by Noggling — APPROVED** [review]

**Pull request merged**